src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -116,59 +116,96 @@ internal interface ISelfCalibratingPredictor
         IPredictor Calibrate(IChannel ch, IDataView data, ICalibratorTrainer caliTrainer, int maxRows);
     }
 
-    public abstract class CalibratedPredictorBase :
+    /// <summary>
+    /// <see cref="IWeaklyTypedCalibratedModelParameters"/> provides a weekly-typed way to access strongly-typed
+    /// <see cref="CalibratedModelParametersBase{TSubPredictor, TCalibrator}.SubModel"/> and
+    /// <see cref="CalibratedModelParametersBase{TSubPredictor, TCalibrator}.Calibrator"/>.
+    /// <see cref="IWeaklyTypedCalibratedModelParameters"/> is commonly used in weekly-typed expressions. The
+    /// existence of this interface is just for supporting existing codebase, so we discourage its uses.
+    /// </summary>
+    [BestFriend]
+    internal interface IWeaklyTypedCalibratedModelParameters
+    {
+        IPredictorProducing<float> WeeklyTypedSubModel { get; }
+        ICalibrator WeeklyTypedCalibrator { get; }
+    }
+
+    /// <summary>
+    /// Class for allowing a post-processing step, defined by <see cref="Calibrator"/>, to <see cref="SubModel"/>'s
+    /// output. For example, in binary classification, <see cref="Calibrator"/> can convert support vector machine's
+    /// output value to the probability of belonging to the positive (or negative) class. Detailed math materials
+    /// can be found at <a href="https://www.csie.ntu.edu.tw/~cjlin/papers/plattprob.pdf">this paper</a>.
+    /// </summary>
```

Comment:
can you put it in remarks?
I don't see much value in links to pdf in intellisence tooltip.

Reply:
Ok.

---
In reply to: [255176861](https://github.com/dotnet/machinelearning/pull/2435#discussion_r255176861) [](ancestors = 255176861)